### ISSUE TITLE
`rememberReferer` does not always seem to be performing correctly

### ISSUE THREAD

**Issue by qq99** [Shopify/turbograft#96]
If you're using a `tg-remote`, I believe it will always call `Turbolinks.pushState` (https://github.com/Shopify/turbograft/blob/master/lib/assets/javascripts/turbograft/turbolinks.coffee#L271)

That is to say, the check for `current url !== referer` will always pass, in most cases.

1.  Perform a regular navigation from A to B
2.  If you break in `reflectNewUrl`, you can see that `referer` was A, and the new page is B (this is the good case, all as expected)
3.  On page B, you click a button that looks like this:
```html
<a tg-remote="POST" href="/resource/1/some_action">do something and refresh this page</a>
```
For this, `some_action` renders `show` but doesn't redirect you anywhere.
4. In your break point, since we haven't performed a regular navigation, `referer` hasn't been updated and is still A.
5. The exact same URL, B, is pushed onto the history stack, and the back button brings you to the same page you were just on

I believe that `tg-remote` should call `rememberReferer` (https://github.com/Shopify/turbograft/blob/master/lib/assets/javascripts/turbograft/turbolinks.coffee#L281-L282) at some point.  Perhaps before it makes the XHR, or perhaps after the `success` of the XHR but before calling `Page.refresh`?

`rememberReferer` isn't currently a public method

cc @Shopify/tnt 

Investigating this, it reminded me of https://github.com/Shopify/turbograft/pull/74 which is similar insofar as they both deal with history.  The easy no-conflict solution would be to provide an attribute that let's you do `tg-no-historystate` or something and leverage the code from #74, but I wonder if we can't just `rememberReferer` with no problems?

Thoughts?  Any potential side effects?

**Comment by qq99** [comment]
Actually, just realized maybe I can add a browser test instead of a unit test

**Comment by TheMallen** [comment]
This was fixed in #97, closing.
